test/cli/autocorrect-kwsplat/test.out

```diff
@@ -0,0 +1,118 @@
+test.rb:11: Passing a non-literal `Hash` to `Object#foo`, which requires specific keyword parameters https://srb.help/7019
+    11 |foo(0, 1, 2, opts)
+                     ^^^^
+    test.rb:4: Keyword parameters of `Object#foo` begin here:
+     4 |sig {params(pos: Integer, x: Integer).void}
+                                  ^
+  Got `T::Hash[Symbol, Integer]` originating from:
+    test.rb:8:
+     8 |opts = T::Hash[Symbol, Integer].new
+               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  Note:
+    Sorbet cannot statically guarantee that the splatted `Hash` contains all the keys required by `foo`.
+    If that's okay, use `T.unsafe` to silence this error, otherwise pass the required keyword parameters directly.
+  Autocorrect: Done
+    test.rb:11: Replaced with `**T.unsafe(opts)`
+    11 |foo(0, 1, 2, opts)
+                     ^^^^
+
+test.rb:13: Passing a non-literal `Hash` to `Object#foo`, which requires specific keyword parameters https://srb.help/7019
+    13 |foo(0, 1, 2, **opts)
+                     ^^^^^^
+    test.rb:4: Keyword parameters of `Object#foo` begin here:
+     4 |sig {params(pos: Integer, x: Integer).void}
+                                  ^
+  Got `T::Hash[Symbol, Integer]` originating from:
+    test.rb:8:
+     8 |opts = T::Hash[Symbol, Integer].new
+               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    test.rb:13:
+    13 |foo(0, 1, 2, **opts)
+                     ^^^^^^
+  Note:
+    Sorbet cannot statically guarantee that the splatted `Hash` contains all the keys required by `foo`.
+    If that's okay, use `T.unsafe` to silence this error, otherwise pass the required keyword parameters directly.
+  Autocorrect: Done
+    test.rb:13: Replaced with `T.unsafe(opts)`
+    13 |foo(0, 1, 2, **opts)
+                       ^^^^
+
+test.rb:15: Passing a non-literal `Hash` to `Object#foo`, which requires specific keyword parameters https://srb.help/7019
+    15 |foo(0, 1, 2, **opts, x: 0)
+                     ^^^^^^^^^^^^
+    test.rb:4: Keyword parameters of `Object#foo` begin here:
+     4 |sig {params(pos: Integer, x: Integer).void}
+                                  ^
+  Got `T::Hash[T.untyped, T.untyped]` originating from:
+    test.rb:15:
+    15 |foo(0, 1, 2, **opts, x: 0)
+                     ^^^^^^^^^^^^
+  Note:
+    Sorbet cannot statically guarantee that the splatted `Hash` contains all the keys required by `foo`.
+    If that's okay, use `T.unsafe` to silence this error, otherwise pass the required keyword parameters directly.
+  Autocorrect: Done
+    test.rb:15: Replaced with `T.unsafe({opts, x: 0})`
+    15 |foo(0, 1, 2, **opts, x: 0)
+                       ^^^^^^^^^^
```

Comment:
> Do we want to highlight all relevant kwargs (as this PR currently shows) for the error message about passing a non-literal hash, or do we want to highlight just the kwsplat arg? The latter seems better to me.

Unfortunately, we don't have that location right now. Note that there could technically be multiple of them:

```ruby
f(**opts1, **opts2, x: nil)
```

We only track one kwsplat for the whole function, and I don't want to try to re-parse it from within calls.cc

> This is a weird case, but the autocorrect suggested by Sorbet is not doing the right thing: it should be T.unsafe({**opts, x: 0}) instead of T.unsafe({opts, x: 0}).

Thanks, good catch. I'll think about that.

Reply:
(the current iteration I fixes this)